### ISSUE TITLE
User mediated credential access limits user agent UI

### ISSUE THREAD

**Issue by dbates-wk** [w3c/webappsec-credential-management#94]
Although section[ Credential Selection](https://w3c.github.io/webappsec-credential-management/#user-mediated-selection) uses passive language for the most part to describe the UI a user agent should show, there is very little room, if any, for making the login experience  look-and-feel like anything other than what is described in that section or illustrated in the non-normative mock chooser, <https://w3c.github.io/webappsec-credential-management/mock-chooser.png>. That is, the look-and-feel of the mock chooser and the following quote are effectively normative:

> SHOULD take the form of a credential chooser which presents the user with a list of credentials that are available for use on a site, allowing them to select one which should be provided to the website, or to reject the request entirely. 

I understand that benefit of such a chooser is to prevent spoofing. However it accomplishes this by effectively mandating a certain look-and-feel that seems jarring and strictly worse than the security value that it brings. From a web developer perspective, it cedes control of the look-and-feel of the login UI to the user agent and hence prevents standardization of UI across all browsers.  For comparison, Safari presents the "credential chooser" as a pop-up menu that is attached to the password field.

Is it necessary for the spec. to effectively mandate UI? What UI would offer the best user experience? Can the spec. be amended so that we can achieve a good user experience?

**Comment by dbates-wk** [comment]
CC @mikewest

**Comment by mikewest** [comment]
I agree that we shouldn't be mandating UX in the document, and I'm happy to make changes to that effect. I had hoped that the language was generic enough to do that, but if y'all have ideas that wouldn't fly with the current language, we should change it. What would you like the spec to say?

**Comment by mikewest** [comment]
CC @battre

**Comment by dbates-wk** [comment]
CC @johnwilander

**Comment by mikewest** [comment]
Ping?
